File: BookWhiz/src/main/java/com/example/BookWhiz/model/Book.java
package com.example.BookWhiz.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;


@Setter
@Getter
@Entity
@Table(name = "book")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "book_author",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "author_id")
    )
    private Set<Author> authors = new HashSet<>();

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "book_genre",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id")
    )
    private Set<Genre> genres =  new HashSet<>();

    @ManyToMany(mappedBy = "books")
    private Set<UserBookList> listsWhereBookBelongs = new HashSet<>();

    @OneToMany(mappedBy="book")
    private Set<Review> reviews = new HashSet<>();

    private String isbn10;

    private String isbn13;

    private String title;

    private String publishingDate;

    private String publishingHouse;

    private String language;

    private String pageCount;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String summary;

    public Book() {}

    public Book(String isbn10, String isbn13, String title, String publishingDate, String publishingHouse, String summary) {
        this.isbn10 = isbn10;
        this.isbn13 = isbn13;
        this.title = title;
        this.publishingDate = publishingDate;
        this.publishingHouse = publishingHouse;
        this.summary = summary;
    }

    public Book(String title, String publishingYear) {
        this.title = title;
        this.publishingDate = publishingDate;
    }

    public String getBookInfo() {

        String bookAuthors = authors.stream()
                .map(Author::getName)
                .collect(Collectors.joining(", "));

        String bookGenres = genres.stream()
                .map(Genre::getName)
                .collect(Collectors.joining(", "));

        return String.format(
                "Title: %s\nISBN13: %s\nPublishing Year: %s\nAuthors: %s\nGenres: %s",
                title,
                isbn13,
                publishingDate,
                bookAuthors,
                bookGenres
        );
    }

}

File: book_whiz/lib/models/book.dart
import 'package:book_whiz/models/author.dart';

import 'genre.dart';

class Book {
  final int? id;
  final String title;
  final String? imageLink;
  final List<Author>? authors;
  final List<Genre>? genres;
  final String? isbn10;
  final String? isbn13;
  final String? publishingDate;
  final String? publishingHouse;
  final String? language;
  final String? pageCount;
  final String? description;


  Book({
    this.id,
    required this.title, // Required field
    this.imageLink,       // Optional field
    this.authors,         // Optional field
    this.genres,          // Optional field
    this.isbn10,          // Optional field
    this.isbn13,          // Optional field
    this.publishingDate,  // Optional field
    this.publishingHouse, // Optional field
    this.language,        // Optional field
    this.pageCount,
    this.description// Optional field
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      imageLink: json['imageLink'],
      authors: List<Author>.from(
          json['authors'].map((authorJson) => Author.fromJson(authorJson))
      ),
      genres: List<Genre>.from(
          json['genres'].map((genreJson) => Genre.fromJson(genreJson))
      ),
      isbn10: json['isbn10'],
      isbn13: json['isbn13'],
      publishingDate: json['publishingDate'],
      publishingHouse: json['publishingHouse'],
      language: json['language'],
      pageCount: json['pageCount'],
      description: json['description']

    );
  }
}